common/workunit/workunit.hpp

```diff
@@ -1688,8 +1690,10 @@ extern WORKUNIT_API bool isValidPriorityValue(const char * priority);
 extern WORKUNIT_API bool isValidMemoryValue(const char * memoryUnit);
 
 #define HourToSeconds(n) ((n)/3600)
-#define NanoSecondsToSeconds(n) ((double)(n)/1000000000)
-inline __int64 calcCost(double ratePerHour, __int64 timeNS) { return HourToSeconds(ratePerHour) * NanoSecondsToSeconds(timeNS) * 1e6; }
+#define ns2Secs(n) ((double)(n)/1000000000)
+#define ms2Secs(n) ((double)(n)/1000)
+inline cost_type calcCost(cost_type ratePerHour, double secs) { return HourToSeconds(ratePerHour) * secs; }
```

Comment:
HoursToSeconds is introducing a potentially significant rounding error.  E.g. if the cost was $0.001/hour - it would round to zero.